src/Data/LocalStorageComponent.ts

```diff
@@ -1,7 +1,30 @@
+import compareVersions from 'compare-versions';
 import { Component } from 'react';
 import LocalStorage from './LocalStorage';
+import { version } from '../version';
 
 abstract class LocalStorageComponent<P = {}, S = {}, SavedState = {}> extends Component<P, S> {
+  public constructor(props: P) {
+    super(props);
+
+    // If the current localstorage was created with a prior version of puzztool, clear
+    // the local storage to get rid of any obsolete or incompatible artifacts
+    const previousVersion = LocalStorage.getObject<string>(this.VERSION_STORAGE_KEY);
+    if (this.versionIncreased(previousVersion, version)) {
+      LocalStorage.clear();
+      // Store the current version
+      LocalStorage.setObject<string>(this.VERSION_STORAGE_KEY, version);
+    }
+  }
+
+  private versionIncreased(prev: string | null, current: string): boolean {
+    if (prev == null) {
+      return true;
```

Comment:
Should this be false?

The way I read this is "if it's your first visit, clear storage"

Reply:
No - if we don't have a stored value, then it means we've upgraded since the last time they've seen.

Case in point - when I tried the latest Morse changes and the Morse page was broken.  In this scenario, if we returned false, then we'd leave the local storage in the broken state.

Reply:
On the very first visit, this will get run the very first time a LocalStorageComponent is used.  At that point, we'll clear the local storage - which has nothing in it at that point - and write the version.  After that, the local storage won't be reset until a new Puzztool version is pushed (or the user clears it from settings)

Reply:
Oh, I see.  Bootstrapping the current non-existence of a version number.

Reply:
I'll add a comment making this clearer.